src/WebApi.OutputCache.V2/CacheOutputAttribute.cs

```diff
@@ -242,12 +255,12 @@ public override void OnActionExecuting(HttpActionContext actionContext)
                         _webApiCache.Add(baseKey, string.Empty, cacheTime.AbsoluteExpiration);
                         _webApiCache.Add(cachekey, content, cacheTime.AbsoluteExpiration, baseKey);
```

Comment:
Wouldn't  it  also run if we are bypassing the cache in 'OnActionExecuting'  by checking if the client requested 'no-cache'? 
Should we then not cache the fresh response at all for that, specific  client call and just return the data we retrieve from executing the requested action and let the existing (stale) cached response be as is,  so that it eventually expires per the expiration policy that was set on the action?